packages/core/crates/core-protocol/src/msg/processor.rs

```diff
@@ -297,8 +297,9 @@ where
                     self.cfg.check_unrealized_balance,
                     &domain_separator,
                 )
-                .await
-                {
+                .await;
+
+                if let Err(e) = validation_res {
                     // Mark as reject and passthrough the error
                     self.db.write().await.mark_rejected(&ticket).await?;
                     return Err(e);
```

Comment:
Is there a way to test it?

Reply:
It was not possible to trigger it with the pending balance check enabled on the sender side. 
Once the sender is allowed to send packets which are not covered by the balance, the relayer is going to encounter this condition (it will not relay). 
Question is whether it should actually also reject the ticket as that goes against what we discussed.